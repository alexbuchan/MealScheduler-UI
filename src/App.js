/* LIBRARY IMPORTS */
import React from "react";
import ReactDOM from "react-dom";
import { BrowserRouter as Router, Switch, Route, Redirect } from "react-router-dom";

/* STORE IMPORTS */
import UserStore from './stores/user';
import ContactStore from './stores/contact';

/* ACTION IMPORTS */
import UserActions from './actions/user/user';

/* VIEW IMPORTS */
import SignupView from './views/SignupView';
import LoginView from './views/LoginView';
import Contacts from './views/Contacts/ContactsView';

/* COMPONENT IMPORTS */
import NavBar from './components/NavBar/NavBar';
import Footer from './components/Footer/Footer';
import GenericNotFound from './components/GenericNotFound/GenericNotFound';
import FlashMessage from './components/FlashMessage/FlashMessage';

/* STYLES IMPORTS */
import './styles/styles.scss';

/* ROOT COMPONENT */
class App extends React.Component {
  constructor() {
    super();

    UserActions.retrieveUserDataOnRefresh();

    this.state = {
      userStore: UserStore.getUserState(),
      closeFlashMessage: UserStore.getCloseFlashMessage(),
      error: '',
      contacts: ContactStore.getContactState().contacts
    }

    this._onChange = this._onChange.bind(this);
  }

  _onChange() {
    this.setState({
      userStore: UserStore.getUserState(),
      error: UserStore.getError(),
      closeFlashMessage: UserStore.getCloseFlashMessage(),
      contacts: ContactStore.getContactState().contacts
    });
  }

  componentDidMount() {
    UserStore.addChangeListener(this._onChange);
    ContactStore.addChangeListener(this._onChange)
  }

  componentWillUnmount() {
    UserStore.removeChangeListener(this._onChange);
    ContactStore.removeChangeListener(this._onChange);
  }

  isLoggedIn = () => {
    if (this.state.userStore.auth) return <Contacts contacts={ this.state.contacts }/>;

    return <Redirect to='/login' />;
  }

  renderFlashMessage() {
    return (
        <FlashMessage
          message={ this.state.error }
          type='error'
          close={ this.state.closeFlashMessage }
          closeFlashMessage={ UserActions.closeFlashMessage }
          timeout={ 3000 }
        />
    );
  }

  render() {
    return (
      <Router>
        <div className='container-fluid'>
          <NavBar 
            user={ this.state.userStore.user }
          />

          <Switch>
            <Route exact path="/" render={ () => <SignupView user={ this.state.userStore.user } /> } />
            <Route exact path="/login" render={ () => <LoginView user={ this.state.userStore.user } /> } />
            <Route exact path="/contacts" render={ this.isLoggedIn } />
            <Route path='/*' component={ GenericNotFound } />
          </Switch>

          { this.renderFlashMessage() }
          <Footer />
        </div>
      </Router>
    );
  }
}

export default App;

ReactDOM.render(<App />, document.getElementById("root"));
