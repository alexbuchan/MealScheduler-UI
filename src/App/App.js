/* LIBRARY IMPORTS */
import React from "react";
import { BrowserRouter as Router, Switch, Route } from "react-router-dom";

/* ACTION IMPORTS */
import UserActions from '../actions/user/UserActions';

/* VIEW IMPORTS */
import Schedule from '../views/Schedule/ScheduleView';
import Recipes from '../views/Recipes/RecipesView';
import Settings from '../views/Settings/Settings';
import GenericNotFound from '../views/GenericNotFoundPage/GenericNotFound';

/* COMPONENT IMPORTS */
import NavBar from '../components/NavBar/NavBar';
import Footer from '../components/Footer/Footer';
import FlashMessage from '../components/FlashMessage/FlashMessage';
import SignupForm from '../components/SignupForm/SignupForm';
import LoginForm from "../components/LoginForm/LoginForm";

/* HOC IMPORTS */
import withUserEntry from '../HOC/UserEntryView/UserEntryView';
import withAuthentication from '../HOC/Authentication/Authentication';

/* STYLES IMPORTS */
import '../styles/styles.scss';

/* ROOT COMPONENT */
class App extends React.Component {
  constructor() {
    super();

    UserActions.retrieveUserDataOnRefresh();
  }

  render() {
    return (
      <Router>
        <div className='container-fluid'>
          <NavBar />

          <Switch>
            <Route exact path="/" component={ withUserEntry(SignupForm) } />
            <Route exact path="/login" component={ withUserEntry(LoginForm) } />
            <Route exact path="/schedule" component={ withAuthentication(Schedule) } />
            <Route exact path="/Recipes" component={ withAuthentication(Recipes) } />
            <Route exact path="/settings" component={ withAuthentication(Settings) } />
            <Route path='/*' component={ GenericNotFound } />
          </Switch>

          <FlashMessage />
          <Footer />
        </div>
      </Router>
    );
  }
}

export default App;
