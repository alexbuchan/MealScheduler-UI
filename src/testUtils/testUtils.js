import React from 'react';
import { shallow } from 'enzyme';

export const componentSetup = (Component, props = {}, options = {}) => {
  const instance = shallow(<Component { ...props } />, options);
  return instance;
}

export const findByTestAttribute = (instance, attribute) => {
  const wrapper = instance.find(`[data-test="${attribute}"]`);
  return wrapper;
}