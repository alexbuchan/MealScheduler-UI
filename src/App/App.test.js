import App from './App';
import { componentSetup } from '../testUtils/testUtils';

describe('App', () => {
  let instance;
  
  beforeEach(() => {
    instance = componentSetup(App);
  });

  it('Should render without errors', () => {
    expect(instance).toMatchSnapshot();
  });
});