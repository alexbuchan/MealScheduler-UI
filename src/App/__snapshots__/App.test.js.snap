// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Should render without errors 1`] = `
<BrowserRouter>
  <div
    className="container-fluid"
  >
    <NavBar />
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/login"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/contacts"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/settings"
      />
      <Route
        component={[Function]}
        path="/*"
      />
    </Switch>
    <FlashMessage />
    <Footer />
  </div>
</BrowserRouter>
`;
